static/js/pm_conversations.js

```diff
@@ -11,6 +12,20 @@ export function is_partner(user_id) {
     return partners.has(user_id);
 }
 
+function filter_muted_pms(conversation) {
+    // This is used to hide muted users from the top left
+    // corner. We don't hide huddles, since they may include
+    // non-muted users and are more similar to streams.
+    if (
+        !conversation.user_ids_string.includes(",") &&
+        muting.is_user_muted(Number.parseInt(conversation.user_ids_string, 10))
+    ) {
+        return false;
+    }
```

Comment:
I think for cases where it's a group PM thread, we might want effectively:
`if all(muting.is_user_muted(user_id) || is_current_user(user_id) for user_id in conversation.user_ids)`

(i.e. we also hide group PM threads if everyone involved is a muted user).

(Though we need to be careful to not have this hide PMs with yourself; worth including that case in our test cases since it's the kinda bug we could avoid now and then add with a future refactor).

Reply:
I've made it so that huddles with all users muted will be hidden, and also expanded the tests to include the self user.

Reply:
This is great and much appreciated, thanks!